scripts/utilities/getCustomPropertyNames.js

```diff
@@ -0,0 +1,28 @@
+const fs = require('fs');
+const path = require('path');
+
+const tokenGroupsDir = path.join(__dirname, '../src/tokens/token-groups');
+
+/**
+ * Allowed Polaris token custom properties.
+ *
+ * Result: ['--p-background', '--p-text', etc...]
+ */
+const getCustomPropertyNames = () => {
+  const polarisTokenCustomProperties = Array.from(
+    new Set(
+      fs
+        .readdirSync(tokenGroupsDir)
+        .map((file) => {
+          const tokenGroup = require(path.join(tokenGroupsDir, file));
+
+          return Object.keys(tokenGroup).map((token) => `--p-${token}`);
+        })
+        .flat(),
+    ),
+  );
+
+  return polarisTokenCustomProperties;
+};
+
+module.exports = getCustomPropertyNames;
```

Comment:
Should this live in the tokens area? I feel like I would expect to find this closer to the JSON files.

Reply:
i think it can go wherever we want. I put it here because this utilities dir already existed and to keep it somewhere "global" so others could access (like the stylelint plugin)